app/scripts/services/hpa.js

```diff
@@ -99,6 +102,27 @@ angular.module("openshiftConsole")
     };
 
 
+    var hasV2HPAAnnotations = function(hpaResources) {
+      return _.some(hpaResources, function(hpa) {
+        return JSON.parse(annotation(hpa, 'autoscaling.alpha.kubernetes.io/metrics'));
+      });
+    };
+
+
+    var hasV2HPAScalerWarning = function(scaleTarget, hpaResources) {
+      var containers = _.get(scaleTarget, 'spec.template.spec.containers', []);
+      var hasRequest = hasCPURequest(containers);
```

Comment:
It's not clear to how this affects whether we show the warning. Even if the container has a request, wouldn't we still show the warning if it's scaling on another metric we don't support in the editor?

Reply:
Well, via the [trello card description](https://trello.com/c/EuQHe0LT) I was under the impression that we are drawing a line between `cpuRequest` and `metrics` for how we show this warning.

Reply:
But, I belive the API does transfer `metrics[]` that are CPU related to a `cpuRequest` if requested via `v1` api, so your comment makes sense.

Reply:
Eliminating this `fn`, then, as we don't need it at all, the `autoscaler` editor does it inline.

Reply:
On that note then, in the editor, we could keep showing the cpu input box for `targetCPUUtilizationPercentage` if that value is set, even if there are other metrics.